types.d.ts

```diff
@@ -42,33 +47,869 @@ declare class CustomerProfileAPI {
+    /**
+     * Get Profile by ID.
+     *
+     * @param {object} [parameters={}] parameters to pass
+     * @returns {Promise<Response>} the response
+     */
+    getProfile(parameters?: any): Promise<Response>;
+    /**
+     * Get ExperienceEvents.
+     *
+     * @param {object} [parameters={}] parameters to pass
+     * @returns {Promise<Response>} the response
+     */
+    getExperienceEvents(parameters?: any): Promise<Response>;
+    /**
+     * Retrieve a list of computed attributes.
+     *
+     * @param {object} [parameters={}] - parameters to pass
+    
+     * @returns {Promise<Response>} the response
+     */
+    listComputedAttributes(parameters?: any): Promise<Response>;
+    /**
+     * Create a computed attribute.
+     *
+     * @param {object} [parameters={}] - parameters to pass
+    
+     * @returns {Promise<Response>} the response
+     */
+    createComputedAttribute(parameters?: any): Promise<Response>;
+    /**
+     * Retrieve a specific computed attribute by its ID.
+     *
+     * @param {object} [parameters={}] - parameters to pass
+     * @param {string} parameters.ATTRIBUTE_ID - The unique, read-only ID of the computed attribute that is generated by the system when the attribute is created.
+     * @returns {Promise<Response>} the response
+     */
+    lookupComputedAttribute(parameters?: {
+        ATTRIBUTE_ID: string;
+    }): Promise<Response>;
+    /**
+     * Update a computed attribute by its ID.
+     *
+     * @param {object} [parameters={}] - parameters to pass
+     * @param {string} parameters.ATTRIBUTE_ID - The unique, read-only ID of the computed attribute that is generated by the system when the attribute is created.
+     * @returns {Promise<Response>} the response
+     */
+    updateComputedAttribute(parameters?: {
+        ATTRIBUTE_ID: string;
+    }): Promise<Response>;
+    /**
+     * Delete a computed attribute by its ID.
+     *
+     * @param {object} [parameters={}] - parameters to pass
+     * @param {string} parameters.ATTRIBUTE_ID - The unique, read-only ID of the computed attribute that is generated by the system when the attribute is created.
+     * @returns {Promise<Response>} the response
+     */
+    deleteComputedAttribute(parameters?: {
+        ATTRIBUTE_ID: string;
+    }): Promise<Response>;
+    /**
+     * Retrieve a list of merge policies.
+     *
+     * @param {object} [parameters={}] - parameters to pass
+     * @param {string} [parameters.x-request-id] - Request Id
+     * @param {string} [parameters.start] - Page offset - As per created time of resource
+     * @param {number} [parameters.limit] - Page Size
+     * @param {string} [parameters.orderBy] - Model attribute which will be using for ordering
+     * @param {string} [parameters.schema.name] - Schema class ID
+     * @param {boolean} [parameters.default] - default flag
+     * @returns {Promise<Response>} the response
+     */
+    getConfigMergePolicies(parameters?: {
+        x-request-id?: string;
```

Comment:
hyphenated variables must be quoted (there are a lot of instances of this that need to be corrected as reported by TSLint. more like a parser issue actually)

I think types.d.ts needs to go through TSLint as well (there is no check in the tooling currently via the template). There are about ~500 TSLint issues. I believe this prevents autocomplete from working (tested it, doesn't show anything).

Reply:
Quotes added if dot or hyphen is in param name.

For TSLint it would be good to solve this with base package, similar to `@adobe/eslint-config-aio-lib-config`.  I can also add TSLint related plugins, but I would need to change global eslintrc config file.